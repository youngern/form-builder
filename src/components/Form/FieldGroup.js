import React from 'react';
import _ from 'lodash';
import { Field } from 'react-final-form';
import Input from './Input';
import Description from './Description';
import Label from './Label';

const FieldGroup = ({ name, placeholder }) => {
  return (
    <>
      <Field
        name={`${name}.name`}
        subscription={{
          value: true,
          active: true,
          touched: true,
          error: true,
        }}>
        {({ input }) => <Label {...input} />}
      </Field>
      <Field
        name={`${name}.description`}
        subscription={{
          value: true,
          active: true,
          touched: true,
          error: true,
        }}>
        {({ input }) => <Description {...input} />}
      </Field>
      <Field
        name={`${name}.initialValue`}
        placeholder={placeholder}
        subscription={{
          value: true,
          active: true,
          touched: true,
          error: true,
        }}>
        {({ input, meta, ...rest }) => {
          return <Input {...input} {...rest} />;
        }}
      </Field>
    </>
  );
};

export default FieldGroup;
