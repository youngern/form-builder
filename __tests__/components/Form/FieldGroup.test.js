import React from 'react';
import { Form } from 'react-final-form';
import FieldGroup from 'final-form-react-native/components/Form/FieldGroup';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Form onSubmit={() => {}}>
        {() => <FieldGroup name="fields[0]" placeholder="placeholder" />}
      </Form>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
