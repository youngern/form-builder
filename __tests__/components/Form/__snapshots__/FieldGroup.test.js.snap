// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#4d80e4",
        "fontFamily": "AppleSDGothicNeo-Bold",
        "fontSize": 14,
      }
    }
  >
    
  </Text>,
  <Text
    style={
      Object {
        "color": "#679b9b",
        "fontFamily": undefined,
        "fontSize": 12,
      }
    }
  >
    
  </Text>,
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="placeholder"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fbfefd",
          "borderColor": "lightgray",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#204051",
          "fontFamily": "AppleSDGothicNeo-Bold",
          "fontSize": 20,
          "paddingLeft": 10,
          "paddingRight": 5,
          "paddingVertical": 10,
        },
        undefined,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />,
]
`;
